python/cudf/cudf/_lib/column.pyx

```diff
@@ -251,8 +255,12 @@ cdef class Column:
 
     @property
     def children(self):
+        if self.base_children is ():
+            return ()
         if self._children is None:
-            self._children = self.base_children
+            self._children = Column.from_unique_ptr(
+                make_unique[column](self.view())
+            ).base_children
```

Comment:
This always copy constructs from `base_children` even if the offset / size are unchanged.

Reply:
Should be fixed now - can you take a look?